ee/server/service/mdm.go

```diff
@@ -586,3 +596,26 @@ func (svc *Service) mdmAppleSyncDEPProfile(ctx context.Context) error {
 
 	return svc.depService.RegisterProfileWithAppleDEPServer(ctx, jsonProf, enrollURL)
 }
+
+func (svc *Service) getAutomaticEnrollmentProfile(ctx context.Context) (*fleet.MDMAppleEnrollmentProfile, error) {
+	profiles, err := svc.ds.ListMDMAppleEnrollmentProfiles(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "listing profiles")
+	}
+
+	// Grab the first automatic enrollment profile we find, the current
+	// behavior is that the last enrollment profile that was uploaded is
+	// the one assigned to newly enrolled devices.
+	//
+	// TODO: this will change after #10995 where there can be a DEP profile
+	// per team.
+	var depProf *fleet.MDMAppleEnrollmentProfile
+	for _, prof := range profiles {
+		if prof.Type == "automatic" {
+			depProf = prof
+			break
+		}
+	}
+
+	return depProf, nil
+}
```

Comment:
I created https://github.com/fleetdm/fleet/issues/11533 to track this and ensure things are in sync before the release.